libsolutil/Assertions.h

```diff
@@ -40,6 +40,27 @@ namespace solidity::util
 #define ETH_FUNC __func__
 #endif
 
+#if defined(__GNUC__)
+// GCC 4.8+, Clang, Intel and other compilers compatible with GCC (-std=c++0x or above)
+[[noreturn]] inline __attribute__((always_inline)) void unreachable()
+{
+    __builtin_unreachable();
+}
+
+#elif defined(_MSC_VER) // MSVC
+
+[[noreturn]] __forceinline void unreachable()
+{
+    __assume(false);
+}
+
+#else
+
+[[noreturn]] inline void unreachable()
+{
+}
```

Comment:
I guess, it'd be safer if this actually did not return, i.e. if it threw a bogus exception... that should then also work for all compilers without case distinction, shouldn't it?

Reply:
right!